docs/book/src/topics/custom-vnet.md

```diff
@@ -30,6 +30,38 @@ If providing an existing vnet and subnets with existing network security groups,
 
 The pre-existing vnet can be in the same resource group or a different resource group in the same subscription as the target cluster. When deleting the `AzureCluster`, the vnet and resource group will only be deleted if they are "managed" by capz, ie. they were created during cluster deployment. Pre-existing vnets and resource groups will *not* be deleted.
 
+Alternatively, pre-existing vnets can be peered with a cluster's newly created vnets by specifying each vnet by name and resource group.
```

Comment:
Let's put this under a new section

## Virtual Network peering

Reply:
let's also add a note at the bottom of the section about how only same subscription vnet peerings are supported currently.

Reply:
And also add a reference/link to this section in the warning that mentions using a peered vnet for private clusters https://capz.sigs.k8s.io/topics/api-server-endpoint.html#warning